src/ui/components/common/sidebar.rs

```diff
@@ -1,135 +1,152 @@
-                } else {
-                    None
-                }}
-              </div>
-              <table class="w-full mt-3 table shadow-lg">
-                <caption class="text-lg font-semibold whitespace-nowrap align-middle text-start mb-2">
-                  <Icon icon=IconType::Users size=IconSize::Large class="inline me-2" />
-                  {move_tr!("active-users")}
-                </caption>
-                <thead>
-                  <tr class="font-extrabold text-sm bg-base-300 *:p-3">
-                    <th class="text-start" scope="col">
-                      {time_frame}
-                    </th>
-                    <th class="text-center" scope="col">
-                      {move_tr!("count")}
-                    </th>
-                  </tr>
-                </thead>
-                <tbody class="bg-base-100">
-                  <UserStatRow text=today count=data.users_today() />
-                  <UserStatRow text=past_week count=data.users_week() />
-                  <UserStatRow text=past_month count=data.users_month() />
-                  <UserStatRow text=past_6_months count=data.users_6_months() />
-                  {match data {
-                      SidebarData::Site { ref counts, .. } => {
-                          view! {
-                            <UserStatRow
-                              text=move_tr!("all-time")
-                              count=counts.users_active_month
-                            />
-                          }
-                              .into_view()
-                      }
-                      SidebarData::Community { ref counts, .. } => {
-                          view! {
-                            <UserStatRow
-                              text=move_tr!("local-subscribers")
-                              count=counts.subscribers_local
-                            />
-                            <UserStatRow text=move_tr!("subscribers") count=counts.subscribers />
-                          }
-                              .into_view()
+        <h2 class="card-title">{name}</h2>
+        {description.map(|description| view! { <p>{description}</p> })}
+        <section aria-labelledby="instance-stats-heading" class="my-4">
+          <h3 id="instance-stats-heading" class="text-2xl font-bold mb-2">
+            {heading}
+          </h3>
+          <div class="font-semibold flex flex-wrap *:m-1.5">
+            <div>
+              <Icon icon=IconType::Posts size=IconSize::Large class="inline" />
+              {posts.pretty_format()}
+              " "
+              <span class="text-sm">{move_tr!("posts")}</span>
+            </div>
+            <div>
+              <Icon icon=IconType::Comments size=IconSize::Large class="inline" />
+              {comments.pretty_format()}
+              " "
+              <span class="text-sm">{move_tr!("comments")}</span>
+            </div>
+
+            {if let SidebarData::Site(s) = data {
```

Comment:
We should probably even make an internal rule, that all views need their own components. It might seem verbose, but then we'd avoid all the mess of massive components like we have with lemmy-ui. 

We'd also get the benefit of seeing a named function that tells you what this block does.

Reply:
> We should probably even make an internal rule, that all views need their own components.

Is a view here referring to a page? And by own components, do you mean just that there is a component for each (like is the case now), or that all pages should have their own components that are specific to them?

> It might seem verbose, but then we'd avoid all the mess of massive components like we have with lemmy-ui.

Breaking repeating things into smaller components is also an option.

Reply:
I just mean if you need to embed a view macro within a view macro, then its probably a good indicator that the inner view, should be its own component.